packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,314 @@
+   * @default false
+   */
+  readonly queryString?: boolean;
+
+  /**
+   * Check if path is to be included inside the execution input
+   * @default false
+   */
+  readonly path?: boolean;
+
+  /**
+   * Check if header is to be included inside the execution input
+   * @default false
+   */
+  readonly headers?: boolean;
+}
+/**
+ * Integrates a Synchronous Express State Machine from AWS Step Functions to an API Gateway method.
+ *
+ * @example
+ *
+ *    const stateMachine = new sfn.StateMachine(this, 'MyStateMachine', ...);
+ *    api.addMethod('GET', new StepFunctionsIntegration(stateMachine));
+ */
+export class StepFunctionsIntegration extends AwsIntegration {
+  private readonly stateMachine: sfn.IStateMachine;
+  private readonly action: string;
+  constructor(stateMachine: sfn.IStateMachine, options: StepFunctionsIntegrationOptions = { action: 'StartSyncExecution' }) {
+
+    const integResponse = integrationResponse();
+    const requestTemplate = requestTemplates(stateMachine, options);
+    super({
+      service: 'states',
+      action: options.action,
+      options: {
+        credentialsRole: options.credentialsRole,
+        integrationResponses: integResponse,
+        passthroughBehavior: PassthroughBehavior.NEVER,
+        requestTemplates: requestTemplate,
+        ...options,
+      },
+    });
+
+    this.stateMachine = stateMachine;
+    this.action = options.action;
+  }
+
+  public bind(method: Method): IntegrationConfig {
+    const bindResult = super.bind(method);
+    const principal = new iam.ServicePrincipal('apigateway.amazonaws.com');
+
+    this.stateMachine.grantExecution(principal, `states:${this.action}`);
+
+    let stateMachineName;
+
+    if (this.stateMachine instanceof sfn.StateMachine) {
+      //if not imported, extract the name from the CFN layer to reach the
+      //literal value if it is given (rather than a token)
+      stateMachineName = (this.stateMachine.node.defaultChild as sfn.CfnStateMachine).stateMachineName;
+    } else {
+      //imported state machine
+      stateMachineName = 'StateMachine-' + (String(this.stateMachine.stack.node.addr));
+    }
+
+    let deploymentToken;
+
+    if (stateMachineName !== undefined && !Token.isUnresolved(stateMachineName)) {
+      deploymentToken = JSON.stringify({ stateMachineName });
+    }
+    return {
+      ...bindResult,
+      deploymentToken,
+    };
+
+  }
+}
+
+/**
+ * Defines the integration response that passes the result, on success,
+ * or the error, on failure, from the backend the client.
+ *
+ * @returns integrationResponse
+ */
+function integrationResponse() {
+  const errorResponse = [
+    {
+      /**
+       * Specifies the regular expression (regex) pattern used to choose
+       * an integration response based on the response from the back end.
+       * In this case it will match all '4XX' HTTP Errors
+       */
+      selectionPattern: '4\\d{2}',
+      statusCode: '400',
```

Comment:
(minor)

rather than convert 429 (say) into 400, should we be returning the same error code? Is that possible/easy to do?

Reply:
I changed to bad request as bad input does not represent the meaning of 400. maybe if we remove this it will behave as expected? not sure.